.pre-commit-config.yaml

```diff
@@ -0,0 +1,108 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: "b0f06dc9f2260909f3423243de180edfc823ec5a"
+    hooks:
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+
+  # - repo: https://github.com/hakancelik96/unimport
+  #   rev: df8eb1a4c91acb84da197828af8157708968b596
+  #   hooks:
+  #     - id: unimport
+  #       args: [--remove, --include-star-import]
+  - repo: https://github.com/pycqa/isort
+    rev: "12cc5fbd67eebf92eb2213b03c07b138ae1fb448"
+    hooks:
+      - id: isort
+        files: "python/.*"
+        args: ["--profile", "black", "--filter-files"]
+
+  - repo: https://github.com/psf/black
+    rev: "6debce63bc2429b1680f8838592f2e56e3df6b27"
+    hooks:
+      - id: black
+
+  # - repo: https://gitlab.com/pycqa/flake8
+  #   rev: "21d3c70d676007470908d39b73f0521d39b3b997"
+  #   hooks:
+  #     - id: flake8
+
+  - repo: https://github.com/kynan/nbstripout
+    rev: 8cafdcc393232045208137698dbeb42d6e0dd9e8
+    hooks:
+      - id: nbstripout
+        files: ".ipynb"
+
+  - repo: https://github.com/asottile/pyupgrade
+    rev: 85f02f45f0d2889f3826f16b60f27188ea84c1ae
+    hooks:
+      - id: pyupgrade
+        args: [--py37-plus, --keep-runtime-typing]
```

Comment:
We might need to change the docs to say that Python 3.7 is required?  And the CI is running 3.6 right now, so this will need to be switched?